servlet-jakarta/src/main/java/com/onelogin/saml2/servlet/jakarta/JakartaSamlHttpRequest.java

```diff
@@ -0,0 +1,62 @@
+package com.onelogin.saml2.servlet.jakarta;
+
+import com.onelogin.saml2.http.HttpRequest;
+import jakarta.servlet.http.HttpServletRequest;
+
+import java.util.Map;
+
+import static com.onelogin.saml2.util.Preconditions.checkNotNull;
+
+public class JakartaSamlHttpRequest implements HttpRequest {
+
+    private final HttpServletRequest delegate;
+
+    private JakartaSamlHttpRequest(HttpServletRequest delegate) {
+        this.delegate = checkNotNull(delegate, "Servlet request cannot be null.");
+    }
+
+    @Override
+    public int getServerPort() {
+        return delegate.getServerPort();
+    }
+
+    @Override
+    public String getScheme() {
+        return delegate.getScheme();
+    }
+
+    @Override
+    public String getServerName() {
+        return delegate.getServerName();
+    }
+
+    @Override
+    public String getRequestURL() {
+        return delegate.getRequestURL().toString();
+    }
+
+    @Override
+    public String getRequestURI() {
+        return delegate.getRequestURI();
+    }
+
+    @Override
+    public String getQueryString() {
+        return delegate.getQueryString();
+    }
+
+    @Override
+    public Map<String, String[]> getParameterMap() {
+        return delegate.getParameterMap();
+    }
+
+    @Override
+    public void invalidateSession() {
+        delegate.getSession().invalidate();
+    }
+
+    public static HttpRequest makeHttpRequest(HttpServletRequest delegate) {
```

Comment:
I'm not even sure this static method is really needed, I mean is this really any better `new JakartaSamlHttpRequest(delegate)`?